src/haversine.jl

```diff
@@ -0,0 +1,30 @@
+"""
+    Haversine(radius)
+
+The haversine distance between two locations on a sphere of given `radius`.
+
+Locations are described with longitude and latitude in degrees.
+The computed distance has the same units as that of the radius.
+"""
+struct Haversine{T<:Real} <: Metric
+    radius::T
+end
+
+function evaluate(dist::Haversine{T}, x::AbstractVector, y::AbstractVector) where {T<:Real}
```

Comment:
Perhaps check that the length is 2? Also, I think it would be nice to accept tuples of length 2 as well.

Reply:
Can this check be performed without performance penalties? That is the main reason I didn't write a test. Also, could you please clarify the interface with tuples you have in mind?

Reply:
My thinking was that if you have a bunch of longitudes and latitudes in a list you would have to preallocate an array to send into this function in order to not cause an allocation at every call. You could use a static array but then you need a package. With a tuple, you won't have any allocations.

Regarding performance time, a check like this should be completely negligible. But you could of course benchmark it.

Reply:
@KristofferC is this issue about pre-allocation specific to Haversine or to the Distances.jl API in general? I didn't write a method with tuples because I didn't see any with the other metrics defined in the package.

Regarding the check for 2D vectors, I prefer to not implement it because I am expecting to call this function some hundreds of thousands of times, I had a performance boost on my package already by just eliminating basic assertions. A user of this distance will know that it is only supposed to make sense with 2D vectors.

Reply:
If you call this function hundreds of thousands of times, the check will have cost you ~0.0001 seconds. However, if you do the length check you can add `@inbounds` on the array access, giving you back that precious nanosecond every call.

Furthermore, if you allow this function to take tuples, you will gain 2 more nanoseconds per call compared to the arrays.

To summarize:

* Doing the check makes the function faster at no performance cost.
* Since you are so worried about performance, allowing tuples will allow you to call this function with better performance.

Reply:
@KristofferC I appreciate your suggestions, but the use case with tuples is not very appealing to me at the moment. All my use cases involve arrays coming from other calculations that aren't necessarily 2D. So specializing on 2D tuples is not a common pattern for me. Also, by grepping the source code of Distances.jl, the only other type using tuples is the RenyDivergence. All other metrics have their implementations with arrays.

Thank you again for your suggestions, but I think a version of Haversine with tuples could be added in the future if someone misses it.

Reply:
Please do, I am not sure I understand what your suggestions are here, but if you can update the PR to be more efficient and satisfy a general interface (that also works with arrays), that would be great.